modules/data/src/main/scala/shapeless3/data/monoidal.scala

```diff
@@ -18,11 +18,9 @@ package shapeless3.data
 
 import scala.compiletime._
 
-import Monoidal._
-
 trait Monoidal {
-  type to[_]
-  type from[_]
+  type to[_] <: Tuple
+  type from[_ <: Tuple]
```

Comment:
Does it only work with `Tuple` now? 🤔

Reply:
I’m not sure, this change was applied in commit 5c4777fbe0f0ee29f1f048acaac300f41df315bf, maybe @milessabin knows more?

Reply:
Tuples are now used as the canonical representation ... there doesn't seem to be any obvious reason to invent another type to play that role.

Reply:
Ok for tuples probably not, but for `Either` the jury is still out I think 🤔

Reply:
No, this is fine for `Either` as well ... see [here](https://github.com/milessabin/shapeless/blob/shapeless-3/modules/data/src/main/scala/shapeless3/data/monoidal.scala#L141-L164).